frontend/src/metabase/dashboard/actions/navigation.js

```diff
@@ -11,15 +11,16 @@ import { getNewCardUrl } from "./getNewCardUrl";
 export const EDIT_QUESTION = "metabase/dashboard/EDIT_QUESTION";
 export const editQuestion = createThunkAction(
   EDIT_QUESTION,
-  question => (dispatch, getState) => {
-    const dashboardId = getDashboardId(getState());
-    const { isNative } = Lib.queryDisplayInfo(question.query());
-    const mode = isNative ? "view" : "notebook";
-    const url = Urls.question(question.card(), { mode });
+  (question, as = "notebook") =>
```

Comment:
I think `as` should be called `mode`.
We can rename the current `mode` to `finalMode` or `effectiveMode`, etc.